pkg/smokescreen/smokescreen.go

```diff
@@ -932,7 +935,18 @@ func checkACLsForRequest(config *Config, req *http.Request, destination hostport
 		return decision
 	}
 
-	ACLDecision, err := config.EgressACL.Decide(role, destination.Host)
+	// X-Upstream-Https-Proxy is a header that can be set by the client to specify
+	// a _subsequent_ proxy to use for the CONNECT request. This is used to allow traffic
+	// flow as in: client -(TLS)-> smokescreen -(TLS)-> external proxy -(TLS)-> destination.
+	// Without this header, there's no way for the client to specify a subsequent proxy.
+	var connectProxyHost string
+	if len(req.Header["X-Upstream-Https-Proxy"]) > 0 {
```

Comment:
Instead of indexing like this directly into the header map could we use https://pkg.go.dev/net/http#Header.Get

Reply:
Fixed!